src/FuncSharp.Examples/ITry.cs

```diff
@@ -0,0 +1,199 @@
+            multiplicationResult.Match(
+                result => Console.Write($"Division successful, result is: {result}."),
+                _ => Console.Write("Division failed, must have divided by zero.")
+            );
+            multiplicationResult.Match(result => Console.Write($"Division successful, result is: {result}."));
+            multiplicationResult.Match(ifSecond: _ => Console.Write("Division failed, must have divided by zero."));
+        }
+
+        private void GettingTryValue(decimal number, decimal multiplier)
+        {
+            ITry<decimal, NetworkOperationError> multiplicationResult = PerformNetworkOperation(_ => number * multiplier);
+            ITry<decimal, Exception> iTryWithExceptionAsError = multiplicationResult.MapError(e => new Exception(e.ToString()));
+
+            // Get method will throw an exception in case of empty option. Using it is an anti-pattern.
+            // You should rather use Match to branch your code into individual cases where each case is guaranteed to work.
+            decimal valueOrExceptionThrown1 = iTryWithExceptionAsError.Get();
+
+            // You can also configure the exception that is thrown by mapping the error inside Get directly.
+            decimal valueOrExceptionThrown2 = multiplicationResult.Get(e => new Exception(e.ToString()));
+            decimal valueOrExceptionThrown3 = iTryWithExceptionAsError.Get(ex => new Exception("Error when multiplying", innerException: ex));
+
+            // Because ITry is a coproduct, you can check the value directly. On ITry, there are named properties for this.
+            // See IOption usages for more information about them.
+            IOption<decimal> successResult = multiplicationResult.Success;
+            IOption<decimal> successResult2 = multiplicationResult.First;
+            IOption<NetworkOperationError> errorResult = multiplicationResult.Error;
+            IOption<NetworkOperationError> errorResult2 = multiplicationResult.Second;
+        }
+
+        public void HandlingExceptionsWithCatch(decimal number, decimal divisor)
+        {
+            // Catches any exception
+            ITry<decimal, Exception> divisionHandlingAllExceptions1 = Try.Catch<decimal, Exception>(_ => number / divisor);
+
+            // Only catches a specific exception.
+            ITry<decimal, DivideByZeroException> divisionHandlingDividingByZero = Try.Catch<decimal, DivideByZeroException>(_ => number / divisor);
+
+            // Catch also has an overload which allows recovering in case of exception.
+            decimal divisionResult = Try.Catch<decimal, DivideByZeroException>(
+                _ => number / divisor,
+                exception => 0
+            );
+        }
+
+        public void HandlingExceptionsWithCreate(decimal number, decimal divisor)
+        {
+            // Catches any exception
+            ITry<decimal> divisionHandlingAllExceptions1 = Try.Create<decimal, Exception>(_ => number / divisor);
```

Comment:
It's not really clear what the difference between Catch and Create is.

Reply:
That's why I don't use `var` but the exact type, so people can see that. But I guess I'll add a comment as well.